src/Stryker.Core/Stryker.Core/Initialisation/ProjectMutator.cs

```diff
@@ -60,7 +69,49 @@ private void EnrichTestProjectsWithTestInfo(InitialTestRun initialTestRun, TestP
                 }
                 else
                 {
-                    _logger.LogDebug("Could not locate unit test in any testfile. This should not happen and results in incorrect test reporting.");
+                    //Test if you can find the file by scanning the sources for testcase name
+                    var qualifiedNameArray = unitTest.FullyQualifiedName.Split('.');
+                    var methodName = qualifiedNameArray[^1];
+                    var className = qualifiedNameArray[^2];
+                    var nameSpace1 = new ArraySegment<string>(qualifiedNameArray, 0, qualifiedNameArray.Length - 2);
+                    var nameSpace = $"namespace {string.Join('.', nameSpace1)}";
+
+                    testFile = testProjectsInfo.TestFiles.Where(tFile => !tFile.FilePath.EndsWith("GlobalSuppressions.cs")).SingleOrDefault(tFile =>
+                        tFile.Source.Contains(className) && tFile.Source.Contains(methodName) && tFile.Source.Contains(nameSpace));
+                    if (testFile is not null)
+                    {
+                        var testDescriptions =
+                            initialTestRun.Result.VsTestDescriptions.Where(td => td.Description.Name == unitTest.FullyQualifiedName);
+                        foreach (var testDescription in testDescriptions)
+                        {
+                            testDescription.Description.TestFilePath = testFile.FilePath;
+                        }
+
+                        var lineNumber = unitTest.LineNumber;
+                        if (lineNumber < 1)
+                        {
+                            var lines = testFile.Source.Split("\r\n");
+                            foreach (var line in lines)
+                            {
+                                if (line.Contains(methodName) && !line.Contains("class"))
+                                {
+                                    lineNumber = Array.IndexOf(testFile.Source.Split("\r\n"),
+                                        testFile.Source.Split("\r\n").First(sourceLine => sourceLine.Contains(methodName) && !sourceLine.Contains("class")));
+                                    break;
+                                }
+                            }
+                        }
+
+                        var lineSpan = testFile.SyntaxTree.GetText().Lines[lineNumber].Span;
+                        var nodesInSpan = testFile.SyntaxTree.GetRoot().DescendantNodes(lineSpan);
+                        var node = nodesInSpan.First(n => n is MethodDeclarationSyntax);
+                        testFile.AddTest(unitTest.Id, unitTest.FullyQualifiedName, node);
+                    }
+                    else
+                    {
+                        _logger.LogDebug(
+                            "Could not locate unit test in any testfile. This should not happen and results in incorrect test reporting.");
+                    }
                 }
```

Comment:
Could be please move this logic to a dedicated class ? at least dedicated methods?

This code needs unit tests.